ee/vellum_ee/workflows/display/nodes/vellum/merge_node.py

```diff
@@ -46,3 +50,18 @@ def serialize(self, display_context: WorkflowDisplayContext, **kwargs: Any) -> J
 
     def get_target_handle_ids(self) -> Optional[List[UUID]]:
         return self._get_explicit_node_display_attr("target_handle_ids", List[UUID])
+
+    def get_target_handle_id_by_source_node_id(self, source_node_id: UUID) -> UUID:
+        target_handle_ids = self.get_target_handle_ids()
+        if target_handle_ids is None:
+            return uuid4_from_hash(f"{self.node_id}|target_handle|{source_node_id}")
+
```

Comment:
dragons below

Reply:
After about 30 min of poking around, I couldn't come up with a more elegant way of handling this now. I think you're right that merge nodes should be one of the first to be cut over to generic nodes.

I've pushed a commit that at least reduces the blast radius of the mess that merge nodes introduce.